pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/minion/generator/BaseTaskGenerator.java

```diff
@@ -120,4 +123,23 @@ public List<PinotTaskConfig> generateTasks(TableConfig tableConfig, Map<String,
       throws Exception {
     throw new UnknownTaskTypeException("Adhoc task generation is not supported for task type - " + this.getTaskType());
   }
+
+  @Override
+  public void generateTasks(List<TableConfig> tableConfigs, List<PinotTaskConfig> pinotTaskConfigs)
+      throws Exception {
+    pinotTaskConfigs.addAll(generateTasks(tableConfigs));
+  }
+
+  @Override
+  public String getMinionInstanceTag(TableConfig tableConfig) {
+    TableTaskConfig tableTaskConfig = tableConfig.getTaskConfig();
+    if (tableTaskConfig != null) {
+      Map<String, String> configs = tableTaskConfig.getConfigsForTaskType(getTaskType());
+      if (!configs.isEmpty()) {
+        return configs.getOrDefault(PinotTaskManager.MINION_INSTANCE_TAG_CONFIG,
```

Comment:
Since the user has to manually fill this in, can we add table config validation to ensure that the minion tag configured here has actually been created? 

If there are typos, it'll be some work to troubleshoot this, if not validated during config time.

Reply:
It's difficult to add in TableConfigUtils as we don't have HelixManager present in that class.

Reply:
I see. It would be ideal to enforce some validation especially given that user has to enter this for every taskConfig.  Im curious if we have any validation for table tenant.
Can we keep track of this work?

Reply:
Actually we do in PinotHelixResourceManager class - https://github.com/apache/pinot/blob/af0e5b82baf196f1bcea40d9737e3d7926cf9958/pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/PinotHelixResourceManager.java#L1781-L1785

But I think we should refactor to have it in the TableConfigUtils class. Tracking it here: #12698